packages/angular/src/migrations/update-19-6-0/turn-off-dts-by-default.ts

```diff
@@ -0,0 +1,76 @@
+import { type Tree, formatFiles, visitNotIgnoredFiles } from '@nx/devkit';
+import { forEachExecutorOptions } from '@nx/devkit/src/generators/executor-options-utils';
+import { minimatch } from 'minimatch';
+import { tsquery } from '@phenomnomnominal/tsquery';
+
+export default async function (tree: Tree) {
+  visitNotIgnoredFiles(tree, '', (path) => {
+    const webpackConfigGlob = '**/webpack*.config*.{js,ts,mjs,cjs}';
+    const result = minimatch(path, webpackConfigGlob);
+    if (!result) {
+      return;
+    }
+    let webpackConfigContents = tree.read(path, 'utf-8');
+    if (
+      !/withModuleFederationSSR|withModuleFederation/.test(
+        webpackConfigContents
+      )
+    ) {
+      return;
+    }
+
+    const WITH_MODULE_FEDERATION_SELECTOR =
+      'CallExpression:has(Identifier[name=withModuleFederation]),CallExpression:has(Identifier[name=withModuleFederationForSSR])';
+    const EXISTING_MF_OVERRIDES_SELECTOR = 'ObjectLiteralExpression';
+
+    const ast = tsquery.ast(webpackConfigContents);
+    const withModuleFederationNodes = tsquery(
+      ast,
+      WITH_MODULE_FEDERATION_SELECTOR,
+      { visitAllChildren: true }
+    );
+    if (!withModuleFederationNodes.length) {
+      return;
+    }
+
+    const withModuleFederationNode = withModuleFederationNodes[0];
+    const existingOverridesNodes = tsquery(
+      withModuleFederationNode,
+      EXISTING_MF_OVERRIDES_SELECTOR,
+      { visitAllChildren: true }
+    );
+    if (!existingOverridesNodes.length) {
+      // doesn't exist, add it
+      webpackConfigContents = `${webpackConfigContents.slice(
+        0,
+        withModuleFederationNode.getEnd() - 1
+      )},${JSON.stringify({ dts: false })}${webpackConfigContents.slice(
+        withModuleFederationNode.getEnd() - 1
+      )}`;
+    } else {
+      let existingOverrideNode;
```

Comment:
We should check if the object already has `dts` property. So if someone has `{ dts: true } ` or `{ dts: false }` they don't get `{ dts: true, dts: false}` or `{ dts: false, dts: false }`